File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/PlayerGameModeChangeEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.Objects;

public final class PlayerGameModeChangeEvent extends AbstractCancellableEvent {

    private final CellPlayer player;
    private final String fromGameMode;
    private final String toGameMode;

    public PlayerGameModeChangeEvent(
            CellPlayer player,
            String fromGameMode,
            String toGameMode
    ) {
        this.player = Objects.requireNonNull(player, "player");
        this.fromGameMode = Objects.requireNonNull(fromGameMode, "fromGameMode");
        this.toGameMode = Objects.requireNonNull(toGameMode, "toGameMode");
    }

    public CellPlayer player() {
        return player;
    }

    public String fromGameMode() {
        return fromGameMode;
    }

    public String toGameMode() {
        return toGameMode;
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/PlayerGameModeChangeEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer

public class PlayerGameModeChangeEvent(
    private val player: CellPlayer,
    private val fromGameMode: String,
    private val toGameMode: String
) : AbstractCancellableEvent() {

    public fun player(): CellPlayer = player

    public fun fromGameMode(): String = fromGameMode

    public fun toGameMode(): String = toGameMode

}
